azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -238,10 +272,55 @@ private void createAttachmentFile() {
     attachmentFileName = file.getAbsolutePath();
   }
 
+  private void attachKafkaAppender(KafkaLog4jAppender appender) {
+    // If present, remove the existing kafka appender
+    if (kafkaAppender.isPresent()) {
+      removeAppender(kafkaAppender);
+      flowLogger.info("Removed old Kafka appender for job " + this.jobId);
+    }
+
+    kafkaAppender = Optional.of(appender);
+    logger.addAppender(kafkaAppender.get());
+    logger.setAdditivity(false);
+    flowLogger.info("Attached new Kafka appender for job " + this.jobId);
+  }
+
+  private KafkaLog4jAppender createKafkaAppender() {
+    KafkaLog4jAppender kafkaProducer = new KafkaLog4jAppender();
+    Props azkProps = AzkabanExecutorServer.getApp().getAzkabanProps();
+    kafkaProducer.setSyncSend(false);
+    kafkaProducer.setBrokerList(azkProps.getString("kafkaLogging.brokerList"));
+    kafkaProducer.setTopic(azkProps.getString("kafkaLogging.topic"));
+    kafkaProducer.setLayout(new EnhancedPatternLayout(
+        "{\"category\": \"%c{1}\","
```

Comment:
I would prefer something like schema to be a resource file since managing this giant string in java will be a pain. What do you think ?